lib/src/facts/collection.cc

```diff
@@ -163,46 +164,50 @@ namespace facter { namespace facts {
             } else {
                 LOG_DEBUG("skipping external facts for \"{1}\": {2}", dir, msg);
             }
-            return found;
+            return;
         }
 
         LOG_DEBUG("searching {1} for external facts.", search_dir);
-
+        external_resolvers_factory erf;
         each_file(search_dir.string(), [&](string const& path) {
-            for (auto const& res : resolvers) {
-                if (res->can_resolve(path)) {
-                    try {
-                        found = true;
-                        res->resolve(path, *this);
-                    }
-                    catch (external::external_fact_exception& ex) {
-                        LOG_ERROR("error while processing \"{1}\" for external facts: {2}", path, ex.what());
-                    }
-                    break;
-                }
+            try {
+                auto resolver = erf.get_resolver(path);
+                files.push_back(make_pair(path, resolver));
+            } catch (external::external_fact_no_resolver& e) {
+                LOG_WARNING("skipping file \"{1}\": {2}", path, e.what());
             }
             return true;
```

Comment:
as method is now void, check if this is still needed

Reply:
still needed, as this return is part of lambda function of `each_file`

Reply:
I think you can drop the `true`